### PR TITLE
fix(connectors): fixes MGDCTRS-754, added query parameter reset_credentials to get addon params to reset client secret

### PR DESCRIPTION
**Pull request by dhirajsb**
## Description
Added query parameter reset_credentials to get addon params to reset client secret

## Verification Steps
Integration tests included must pass.

## Checklist (Definition of Done)
- [X] All acceptance criteria specified in JIRA have been completed
- [X] Unit and integration tests added that prove the fix is effective or the feature works (tested against emulated and non-emulated OCM environment)
~~- [ ] Documentation added for the feature~~
- [ ] CI and all relevant tests are passing
- [ ] Code Review completed
~~- [ ] Verified independently by reviewer~~
~~- [ ] Required metrics/dashboards/alerts have been added (or PR created).~~
~~- [ ] Required Standard Operating Procedure (SOP) is added.~~
~~- [ ] JIRA has created for changes required on the client side~~

### CODE REVIEW

**Comment by dhirajsb** [comment]
@akoserwal does this look like the correct way to reset client secrets?

**Comment by akoserwal** [comment]
> @akoserwal does this look like the correct way to reset client secrets?

yes

**Comment by dhirajsb** [comment]
@lburgazzoli @valdar please approve.

File: internal/connector/internal/handlers/connector_cluster.go [hunk 1]
```diff
@@ -203,11 +203,17 @@ func (h *ConnectorClusterHandler) GetAddonParameters(w http.ResponseWriter, r *h
 		Action: func() (i interface{}, serviceError *errors.ServiceError) {
 
 			// To make sure the user can access the cluster....
-			cluster, err := h.Service.Get(r.Context(), connectorClusterId)
+			cluster, err := h.Service.Get(ctx, connectorClusterId)
 			if err != nil {
 				return nil, err
 			}
 
+			if r.URL.Query().Get("reset_credentials") == "true" {
```
Comment (valdar) [inline internal/connector/internal/handlers/connector_cluster.go:211]: Just a suggestion: would be more true to the intent to call this something like `refresh_credentials` ?

File: internal/connector/internal/services/connector_cluster.go [hunk 2]
```diff
@@ -919,3 +920,20 @@ func (k *connectorClusterService) GetClusterOrg(id string) (string, *errors.Serv
 	}
 	return cluster.OrganisationId, nil
 }
+
+func (k *connectorClusterService) ResetServiceAccount(ctx context.Context, cluster *dbapi.ConnectorCluster) *errors.ServiceError {
+	secret, serr := k.keycloakService.GetKafkaClientSecret(cluster.ClientId)
```
Comment (valdar) [inline internal/connector/internal/services/connector_cluster.go:925]: I assume this is taking care to cleanup the old secret as well?

File: internal/connector/internal/handlers/connector_cluster.go [hunk 3]
```diff
@@ -203,11 +203,17 @@ func (h *ConnectorClusterHandler) GetAddonParameters(w http.ResponseWriter, r *h
...
+				if serviceError = h.Service.ResetServiceAccount(ctx, &cluster); serviceError != nil {
```
Comment (valdar) [inline internal/connector/internal/handlers/connector_cluster.go:212]: In the case of the above suggestion this needs to be renamed accordingly.

File: openapi/connector_mgmt.yaml [hunk 4]
```diff
@@ -691,6 +691,12 @@ paths:
           type: string
         in: path
         required: true
+      - name: reset_credentials
```
Comment (valdar) [inline openapi/connector_mgmt.yaml:694]: In the case of the above suggestion this needs to be renamed accordingly.

File: internal/connector/test/integration/features/connector-agent-api.feature [hunk 5]
```diff
@@ -38,6 +38,16 @@ Feature: connector agent API
     And get and store access token using the addon parameter response as ${shard_token} and clientID as ${clientID}
     And I remember keycloak client for cleanup with clientID: ${clientID}
 
+    # test client credentials reset
```
Comment (valdar) [inline internal/connector/test/integration/features/connector-agent-api.feature:41]: In the case of the above suggestion this needs to be renamed accordingly.

File: openapi/connector_mgmt.yaml [hunk 6]
```diff
@@ -691,6 +691,12 @@ paths:
...
+        description: Resets cluster service account credentials when true
```
Comment (valdar) [inline openapi/connector_mgmt.yaml:695]: In the case of the above suggestion this needs to be renamed accordingly.

File: internal/connector/internal/services/connector_cluster.go [hunk 7]
```diff
@@ -49,6 +49,7 @@ type ConnectorClusterService interface {
 	CleanupDeployments() *errors.ServiceError
 	ReconcileDeletingClusters() (int, []*errors.ServiceError)
 	GetClusterOrg(id string) (string, *errors.ServiceError)
+	ResetServiceAccount(ctx context.Context, cluster *dbapi.ConnectorCluster) *errors.ServiceError
```
Comment (valdar) [inline internal/connector/internal/services/connector_cluster.go:52]: In the case of the above suggestion this needs to be renamed accordingly.

**Review by valdar — CHANGES_REQUESTED** [review]

File: internal/connector/internal/handlers/connector_cluster.go (hunk 1 above)
Comment (dhirajsb) [reply internal/connector/internal/handlers/connector_cluster.go:211]: No, that's incorrect. It resets the credentials, the token will change. Hence the name.

File: internal/connector/internal/services/connector_cluster.go (hunk 2 above)
Comment (dhirajsb) [reply internal/connector/internal/services/connector_cluster.go:925]: Yeah, the old secret is reset in SSO service. That's what the incorrectly named `GetKafkaClientSecret()` does. That's why I asked @akoserwal to verify.

**Review by valdar — APPROVED** [review]

**Pull request merged**